pkg/pgclient/client.go

```diff
@@ -77,18 +77,25 @@ func getPgConfig(cfg *Config) (*pgxpool.Config, int, error) {
 		return nil, numCopiers, err
 	}
 
-	var pgConfig *pgxpool.Config
+	var (
+		pgConfig          *pgxpool.Config
+		connectionArgsFmt string
+	)
 	if cfg.DbUri == defaultDBUri {
-		pgConfig, err = pgxpool.ParseConfig(connectionStr + fmt.Sprintf(" pool_max_conns=%d pool_min_conns=%d", maxConnections, minConnections))
+		connectionArgsFmt = "%s pool_max_conns=%d pool_min_conns=%d statement_cache_capacity=%d"
 	} else {
-		pgConfig, err = pgxpool.ParseConfig(connectionStr + fmt.Sprintf("&pool_max_conns=%d&pool_min_conns=%d", maxConnections, minConnections))
+		connectionArgsFmt = "%s&pool_max_conns=%d&pool_min_conns=%d&statement_cache_capacity=%d"
 	}
-
+	connectionStringWithArgs := fmt.Sprintf(connectionArgsFmt, connectionStr, maxConnections, minConnections, cfg.MetricsCacheSize)
+	pgConfig, err = pgxpool.ParseConfig(connectionStringWithArgs)
 	if err != nil {
 		log.Error("msg", "configuring connection", "err", util.MaskPassword(err.Error()))
 		return nil, numCopiers, err
 	}
-
+	if cfg.PreferSimpleProtocol {
```

Comment:
Let's add a gauge that exposes this in telemetry. 

Also, can we expose the StatementCache Cap() and Len() through telemetry. Cap() should probably be a single gauge but for Len() we need a histogram of measurements for all the connections we use. We can probably fill this histogram in the pool's AfterRelease callback.